### DISCUSSION TITLE
server problem with router public real ip address

### DISCUSSION THREAD

**Discussion by sabikrahat** [pocketbase/pocketbase#1953] [Q&A] [answered]
Hello!

I'm using pocketbase locally without any online cloud. When I run server in locally `http://127.0.0.1:8090/` it's working properly. Even I run the server publicly using `pocketbase serve --http="0.0.0.0:1010"` it's only work with my device ip address but it doesn't work at my public real ip  address sometime. Firstly it was working without any issue. when I upgrade pocketbase it shows the error. I have already tried with delete all files and setup new service from scratch but same issue. :(

> Router port setup

![router-config](https://user-images.githubusercontent.com/49031412/222242417-4df1cd75-9ec9-4e18-92d3-15394d60e11d.png)

> server running

![serving-config](https://user-images.githubusercontent.com/49031412/222242442-b06bf8d7-b84d-49ee-bace-b9389288e922.png)

> working with my device ipaddress

![local-ip](https://user-images.githubusercontent.com/49031412/222242452-0fb31786-ab06-4fca-b654-f66af2de09a5.png)

> issue with public real ip

![real-ip](https://user-images.githubusercontent.com/49031412/222242461-41921037-54f2-4dde-ab8d-e497be1295c7.png)

**Comment by ganigeorgiev** [comment, marked as answer]
I don't think this is related to PocketBase.

There are too many factors that can result to the above error - misconfigured port forwarding, firewall, some os specific network configuration, reverse proxy misconfiguration (if any), incorrect local ip(s), etc.

Make sure that your local IPs are properly mapped. From the browser screenshot it seems that you are accessing PocketBase at `192.168.10.102` while at the Port forwarding screenshot you are using `192.168.10.101`.

**Comment by sabikrahat** [comment]
Thank you for your suggestion.... Maybe I have done some mistake. Let me check again.

**Comment by oSethoum** [comment]
Hi @sabikrahat, am also facing this problem with pocketbase even after disabling the firewall for both private and public network, have you found a solution for it?

**Reply by sabikrahat** [reply]
The problem was not from pocketbase. Flutter web can't fetch data from HTTP type requests in the release build. You must have to send data using https. But in android you can easily fetch data by using `android:usesCleartextTraffic="true"` in android manifest file.

`
<application
       ...............................
       android:usesCleartextTraffic="true"
       ..........................>
</application>
`

**Reply by oSethoum** [reply]
am not using flutter client, have you tried pocketbase in local private network? apparently am unable to access the admin panel from other PC in the same network using the IP address and the port if the PC running pocketbase, I checked the firewall and everything is fine there, I also tried simple http server and it works fine in he local network.

**Reply by sabikrahat** [reply]
which command did you use to start the server?

**Reply by oSethoum** [reply]
Am using it as a framework here's how I start it
```go
package main

import (
	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/cmd"
)

func main() {
	app := pocketbase.New()
	app.Bootstrap()
	serveCmd := cmd.NewServeCommand(app, true)
	serveCmd.Execute()
}

```

**Reply by oSethoum** [reply]
![image](https://user-images.githubusercontent.com/88779394/222946754-2e4d7375-5a24-4ae5-9b80-b8ec20fb920d.png)
Trying to access it from the machine IP also doesn't work.

**Reply by sabikrahat** [reply]
I don't have enough knowledge about the framework you are using. Maybe you are using the gin-go framework. Try to run the server using this command `pocketbase serve --http="0.0.0.0:<port-number>` example: `pocketbase serve --http="0.0.0.0:8090"`

**Reply by oSethoum** [reply]
Thanks, using this command `pocketbase serve --http="0.0.0.0:<port-number>`  exposed the server to the local network.
@ganigeorgiev  How can I provide this flag using pocketbase as framework?

**Reply by ganigeorgiev** [reply]
@oSethoum If you use directly `NewServeCommand()`, then you have to specify explicitly the `os.Args`, for example:

```go
app := pocketbase.New()
app.Bootstrap()
serveCmd := cmd.NewServeCommand(app, true)
serveCmd.SetArgs([]string{"--http=0.0.0.0:8092"})
serveCmd.Execute()
```

This eventually will change in the future and instead the `serve` arguments will be passed as a config struct.

**Reply by oSethoum** [reply]
Thank you, that works well.
